openedx/core/lib/api/utils.py

```diff
@@ -0,0 +1,57 @@
+import functools
+from importlib import import_module
+
+from django.conf import settings
+
+
+def pluggable_override(override):
+    """
+    This decorator allows overriding any function or method by pointing to an alternative implementation
+    with `override` param.
+    :param override: path to the alternative function
+
+    Usage:
+
+    1. Add this decorator to your original function. `DO_SOMETHING_IMPL` is the variable name in settings that can be
+       used for overriding this function.
+        >>> @pluggable_override('DO_SOMETHING_IMPL')
+        ...     def do_something(self):
```

Comment:
@nedbat, indeed it is, thanks for catching it. I'll update it while addressing other comments.

> Actually, why are these REPL examples? That's not where people will be using this decorator.

I haven't found a better way to include some example code with syntax highlighting directly in the docstring. It might be possible to use the reStructured Text format here, but the syntax highlighting didn't work in my text editors then. I'm open to ideas here.